versioned/tiered/rocksdb/src/main/java/com/dremio/nessie/versioned/store/rocksdb/RocksDBConditionVisitor.java

```diff
@@ -0,0 +1,170 @@
+
+    private boolean isSize(Value value) {
+      return (value.getType() == Value.Type.FUNCTION) && (((ExpressionFunction)value).getName() == ExpressionFunction.FunctionName.SIZE);
+    }
+  }
+
+  /**
+   * This provides a separation of queries on @{ExpressionFunction} from the object itself.
+   * This uses the Visitor design pattern to retrieve object attributes.
+   */
+  private static class RocksDBFunctionHolderValueVisitor implements ValueVisitor<Function> {
+    @Override
+    public Function visit(Value value) {
+      throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Function visit(ExpressionFunction value) {
+      final ExpressionFunction.FunctionName name = value.getName();
+      final List<Value> arguments = value.getArguments();
+      if (arguments.size() != name.getArgCount()) {
+        throw new InvalidParameterException(
+          String.format("Number of arguments provided [%d] does not match the number expected [%d] for %s.",
+            arguments.size(), name.getArgCount(), name));
+      }
+
+      switch (name) {
+        case EQUALS:
+          // Special case SIZE, as the object representation is not contained in one level of ExpressionFunction.
+          if (isSize(arguments.get(0))) {
+            return new Function(Function.SIZE,
+                arguments.get(0).getFunction().getArguments().get(0).accept(STR_VALUE_VISITOR), arguments.get(1).getValue());
+          }
+
+          return new Function(Function.EQUALS,
+            arguments.get(0).accept(STR_VALUE_VISITOR), arguments.get(1).getValue());
+        default:
+          throw new UnsupportedOperationException(String.format("%s is not a supported top-level RocksDB function.", name));
+      }
+    }
+
+    @Override
+    public Function visit(ExpressionPath value) {
+      throw new UnsupportedOperationException();
+    }
+
+    private boolean isSize(Value value) {
+      return (value.getType() == Value.Type.FUNCTION) && (((ExpressionFunction)value).getName() == ExpressionFunction.FunctionName.SIZE);
+    }
+  }
+
+  static final RocksDBStrValueVisitor STR_VALUE_VISITOR = new RocksDBStrValueVisitor();
+  static final RocksDBFunctionHolderValueVisitor FUNC_VALUE_VISITOR = new RocksDBFunctionHolderValueVisitor();
+
+  /**
+  * This is a callback method that ConditionExpression will call when this visitor is accepted.
+  * It creates a ConditionExpressionHolder representation of the ConditionExpression object.
+  * @param conditionExpression to be converted into BSON.
+  * @return the converted ConditionExpression object in BSON format.
+  */
+  @Override
+  public Condition visit(final ConditionExpression conditionExpression) {
+    //    final String functions = conditionExpression.getFunctions().stream()
+    //      .map(f -> f.accept(STR_VALUE_VISITOR))
+    //      .collect(Collectors.joining("& "));
+
+    Condition holder = new Condition();
+    holder.functionList = conditionExpression.getFunctions().stream()
```

Comment:
Have Condition take the list as part of construction to simplify the code here.

Reply:
Agreed. Done.